src/main/java/uk/gov/companieshouse/company/profile/service/CompanyProfileService.java

```diff
@@ -129,4 +133,29 @@ public void updateInsolvencyLink(String contextId, String companyNumber,
         }
 
     }
+
+    private void updateSpecificFields(CompanyProfileDocument companyProfileDocument) {
+        Update update = new Update();
+        setUpdateIfNotNull(update, "data.etag",
+                companyProfileDocument.getCompanyProfile().getEtag());
+        setUpdateIfNotNull(update, "updated",
+                companyProfileDocument.getUpdated());
+        setUpdateIfNotNull(update, "data.links.insolvency",
+                companyProfileDocument.getCompanyProfile().getLinks().getInsolvency());
+        setUpdateIfNotNull(update, "data.links.charges",
```

Comment:
We should delete "links.charges" element if value is null?

Reply:
Same as above